project/npda/views/npda_users.py

```diff
@@ -431,9 +434,68 @@ class NPDAUserDeleteView(
     success_url = reverse_lazy("npda_users")
 
 
-class NPDAUserLogsListView(LoginAndOTPRequiredMixin, ListView):
+class NPDAUserLogsListView(LoginAndOTPRequiredMixin, PermissionRequiredMixin, ListView):
     template_name = "npda_user_logs.html"
     model = VisitActivity
+    permission_required = "npda.view_visitactivity"
+    permission_denied_message = "You do not have the appropriate permissions to access this page/feature. Contact your Coordinator for assistance."
+
+    def has_permission(self):
+        """
+        Custom permission check:
+        - Readers and Editors can only view their own logs.
+        - Coordinators can view logs for any user in their PDU.
+        - RCPCH Audit Team can view logs for any user.
+        """
+        # Get the user whose logs are being viewed
+        npdauser_id = self.kwargs.get("npdauser_id")
+        if not npdauser_id:
+            return False
+
+        try:
+            npda_user = NPDAUser.objects.get(pk=npdauser_id)
+        except NPDAUser.DoesNotExist:
+            return False
+
+        # If user is a Reader or Editor, they can only see their own logs
+        logged_in_user: NPDAUser = self.request.user
+        is_reader_or_editor = logged_in_user.role in [
```

Comment:
For other user related actions we have tended to check permissions directly rather than role. Eg `npda.change_npdauser`.

I'm not sure it's justified to add another permission for this but maybe we check the existing permission instead for consistency?

I'm torn between the options. What do you think?